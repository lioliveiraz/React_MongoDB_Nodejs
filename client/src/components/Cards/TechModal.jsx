import { Card, CardContent } from "@material-ui/core";
import React from "react";
import Typography from "@material-ui/core/Typography";
import { useStyles } from "./../../assets/css/Cards/techModal";

function TechModal({ name, creator, description }) {
  const classes = useStyles();
  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography gutterBottom>Technology</Typography>
        <Typography variant="h5" component="h2" data-cy="modal-title">
          {name}
        </Typography>
        <Typography color="inherit" data-cy="modal-description">
          {description}
        </Typography>
        <Typography variant="body2" component="p" data-cy="modal-creator">
          {creator}
        </Typography>
      </CardContent>
    </Card>
  );
}

export default TechModal;
