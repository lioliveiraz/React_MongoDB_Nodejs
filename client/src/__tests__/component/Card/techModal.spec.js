import React from "react";
import { shallow } from "enzyme";
import TechModal from "./../../../components/Cards/TechModal";
import { validateTruthiness } from "./../../utils/testUtils";

describe("Tech Modal", () => {
  let wrapper;
  const data = {
    name: "name",
    creator: "creator",
    description: "description",
  };

  beforeEach(() => {
    wrapper = shallow(<TechModal {...data} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render correctly", () => {
    const title = wrapper.exists('[data-cy="modal-title"]');
    const description = wrapper.exists('[data-cy="modal-description"]');
    const creator = wrapper.exists('[data-cy="modal-creator"]');

    const elementArr = [title, description, creator];

    elementArr.forEach((el) => {
      validateTruthiness(el);
    });
  });

  it("should render according to props", () => {
    const title = wrapper.find('[data-cy="modal-title"]').text();
    const description = wrapper.find('[data-cy="modal-description"]').text();
    const creator = wrapper.find('[data-cy="modal-creator"]').text();

    expect(title).toEqual(data.name);
    expect(description).toEqual(data.description);
    expect(creator).toEqual(data.creator);
  });
});
