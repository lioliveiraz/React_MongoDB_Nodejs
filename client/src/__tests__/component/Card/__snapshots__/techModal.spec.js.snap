// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tech Modal should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Tech Modal should match snapshot 2`] = `"<div class=\\"MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom\\">Technology</p><h2 class=\\"MuiTypography-root MuiTypography-h5\\" data-cy=\\"modal-title\\">name</h2><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit\\" data-cy=\\"modal-description\\">description</p><p class=\\"MuiTypography-root MuiTypography-body2\\" data-cy=\\"modal-creator\\">creator</p></div></div>"`;
